### DISCUSSION TITLE
Loading partial pre-trained weights for backbone and initialize the rest randomly

### DISCUSSION THREAD

**Discussion by ErikValle** [ultralytics/yolov5#6796] [Q&A] [answered]
Hello. I am trying to train a model using pre-trained weights in the backbone layers but initialize the rest randomly. I saw that excluding the first ten layers (from CONV to SSPF) using the intersect_dicts() function and keeping the random weights generated by the temporal model work [here](https://github.com/ultralytics/yolov5/issues/3293). Nevertheless, I do not know how to declare it in a line as you did it releasing the anchors:

`exclude = ['anchor'] if (opt.cfg or hyp.get('anchors')) and not opt.resume else []  # exclude keys 
`

**Comment by glenn-jocher** [comment]
@ErikValle `exclude` is just a python list. You can add any layer names you do not want transferred to it.

**Reply by ErikValle** [reply]
For instance, I want to exclude layers CONV[0] and CONV[1]. When printing the model, each layer does not have a specific name. The image below shows the previous statement. 

![image](https://user-images.githubusercontent.com/19402707/155905589-3ea3cf27-bd0c-4199-b2b2-11b02597ceea.png)

I tried to identify those weights and biases belonging to CONV[0] and CONV[1] using : `model_dict = model.state_dict()`. Such code retrieves the weights of each layer with labels such as 'model.model.model.0.cv1.conv.weight', 'model.model.model.0.cv1.conv.bias', 'model.model.model.1.cv1.conv.weight', and 'model.model.model.1.cv1.conv.bias'. I have already appended those names in the exclude list as in ['anchor'], but it does not work. On the other hand, if exclude=['anchor', '1'], only 270/745 items are transferred from yolov5x.pt.

Thank you

**Comment by ErikValle** [comment, marked as answer]
This is what I've done to exclude the layers described above:

```
for t in range(0,2):
    exclude.append('model.'+str(t))
```

Such a small code was added after line 125 of the train.py
